### PR TITLE
solution

### PR DESCRIPTION
**Pull request by olenitut**
[DEMO LINK](https://olenitut.github.io/react_movies-list-add-form/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,16 +1,23 @@
+import { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const [neededMovies, setNeededMovies] = useState([...moviesFromServer]);
```
Comment (marina-tilniak) [inline src/App.tsx:9]:
```suggestion
  const [movies, setMovies] = useState([...moviesFromServer]);
```

File: src/components/NewMovie/NewMovie.tsx [hunk 2]
```diff
@@ -1,45 +1,82 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+export const NewMovie:React.FC<{
+  onAdd: (movie: Movie)=>void
```
Comment (marina-tilniak) [inline src/components/NewMovie/NewMovie.tsx:6]: move typing for Props above

File: src/components/NewMovie/NewMovie.tsx [hunk 3]
```diff
@@ -1,45 +1,82 @@
...
+
+  const {
+    title, description, imdbId, imdbUrl, imgUrl,
+  } = userMovie;
+
+  const formIsValid = title && imdbId && imdbUrl && imgUrl;
+
+  const changeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { target } = e;
+    const key = target.id.slice(0, target.id.indexOf('-'));
```
Comment (marina-tilniak) [inline src/components/NewMovie/NewMovie.tsx:27]: just use imdbId

File: src/components/TextField/TextField.tsx [hunk 4]
```diff
@@ -20,7 +20,7 @@ export const TextField: React.FC<Props> = ({
   required = false,
   onChange = () => {},
 }) => {
-  // generage a unique id once on component load
+  // generage a unique idonce on component load
```
Comment (marina-tilniak) [inline src/components/TextField/TextField.tsx:23]: :)

**Review by marina-tilniak — CHANGES_REQUESTED** [review]

File: src/components/NewMovie/NewMovie.tsx [hunk 5]
```diff
@@ -1,45 +1,84 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie)=>void
```
Comment (Anastasiia-Svintsova) [inline src/components/NewMovie/NewMovie.tsx:6]:
```suggestion
  onAdd: (movie: Movie) => void
```

File: src/components/NewMovie/NewMovie.tsx [hunk 6]
```diff
@@ -1,45 +1,84 @@
...
+  };
+
+  const [count, setCount] = useState(0);
+  const [userMovie, setUserMovie] = useState(defaultMovie);
+
+  const {
+    title, description, imdbId, imdbUrl, imgUrl,
+  } = userMovie;
+
+  const formIsValid = title && imdbId && imdbUrl && imgUrl;
```
Comment (Anastasiia-Svintsova) [inline src/components/NewMovie/NewMovie.tsx:25]: use useMemo for this value, please

**Review by Anastasiia-Svintsova — CHANGES_REQUESTED** [review]
- I shouldn't be able to add movies with required empty fields

https://user-images.githubusercontent.com/99403198/211149539-cb461d39-dd8e-4b7f-94c2-f2d5f0ea234c.mov

File: src/components/TextField/TextField.tsx [hunk 7]
```diff
@@ -20,12 +20,10 @@ export const TextField: React.FC<Props> = ({
   required = false,
   onChange = () => {},
 }) => {
-  // generage a unique id once on component load
   const [id] = useState(() => `${name}-${getRandomDigits()}`);
 
-  // To show errors only if the field was touched (onBlur)
   const [touched, setToched] = useState(false);
-  const hasError = touched && required && !value;
+  const hasError = touched && required && !value.trim();
```
Comment (mykhalenych) [inline src/components/TextField/TextField.tsx:26]: You added useMemo for formIsValid, add here as well

File: src/App.tsx [hunk 8]
```diff
@@ -1,16 +1,23 @@
+import { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const [movies, setMovies] = useState([...moviesFromServer]);
```
Comment (mykhalenych) [inline src/App.tsx:9]: `[...moviesFromServer]` the same as `(moviesFromServer)`

**Review by mykhalenych — CHANGES_REQUESTED** [review]

File: src/components/TextField/TextField.tsx (hunk 7 above)
Comment (olenitut) [reply src/components/TextField/TextField.tsx:26]: I added it because the prev mentor asked for it, and only in that place((

**Review by mykhalenych — APPROVED** [review]

**Pull request closed without merging**